map/autoware_lanelet2_map_validator/test/src/test_regulatory_element_details.cpp

```diff
@@ -0,0 +1,368 @@
+    Lanelet cw2 = Lanelet(getId(), cw_ls2, cw_ls4, cw_attr);
+
+    // Traffic light regulatory element
+    RegulatoryElementPtr tl_reg_elem1, tl_reg_elem2;
+    tl_reg_elem1 = TrafficLight::make(getId(), tl_re_attr, {tl1}, {sl1});
+    tl_reg_elem2 = TrafficLight::make(getId(), tl_re_attr, {tl2}, {sl2});
+
+    // Crosswalk regulatory element
+    RegulatoryElementPtr cw_reg_elem1, cw_reg_elem2;
+    cw_reg_elem1 = Crosswalk::make(getId(), cw_re_attr, cw1, cw_poly1, {sl1});
+    cw_reg_elem2 = Crosswalk::make(getId(), cw_re_attr, cw2, cw_poly2, {sl2});
+
+    cw1.addRegulatoryElement(cw_reg_elem1);
+    cw2.addRegulatoryElement(cw_reg_elem2);
+
+    // Add elements to map
+    for (const auto & re : {tl_reg_elem1, tl_reg_elem2}) {
+      in_map_ptr->add(re);
+    }
+    for (const auto & cw : {cw1, cw2}) {
+      in_map_ptr->add(cw);
+    }
+  }
+  AttributeMap sl_attr, tl_attr, cw_attr, cw_poly_attr, tl_re_attr, cw_re_attr;
+
+private:
+};
+
+TEST_F(TestSuite, ValidatorAvailability)  // NOLINT for gtest
+{
+  std::string expected_validators_concat =
+    "mapping.crosswalk.regulatory_element_details,"
+    "mapping.traffic_light.regulatory_element_details";
+
+  lanelet::validation::Strings validators =
+    lanelet::validation::availabeChecks(expected_validators_concat);  // cspell:disable-line
+  uint8_t expected_num_validators = 2;
+  std::cout << "size: " << validators.size() << std::endl;
+  EXPECT_EQ(expected_num_validators, validators.size());
+
+  std::set<std::string> expected_validators_set = {
+    "mapping.crosswalk.regulatory_element_details",
+    "mapping.traffic_light.regulatory_element_details"};
+
+  std::set<std::string> testing_validators_set = {
+    lanelet::validation::RegulatoryElementsDetailsForCrosswalksValidator::name(),
+    lanelet::validation::RegulatoryElementsDetailsForTrafficLightsValidator::name()};
+
+  for (const auto & name : testing_validators_set) {
+    std::cout << name << std::endl;
+    EXPECT_TRUE(expected_validators_set.find(name) != expected_validators_set.end())
+      << "Unexpected validator found: " << name;
+  }
+}
+
+TEST_F(TestSuite, RegulatoryElementOfTrafficLightWithoutTrafficLight)  // NOLINT for gtest
+{
+  // Check regulatory element of traffic light without traffic light
+
+  RegulatoryElementPtr tl_reg_elem_no_tl;
+  // Line string without traffic light attribute
+  const auto & ls = LineString3d(99998, {});
+  LaneletMapPtr test_map_ptr = lanelet::utils::createMap({ls});
+  // Traffic light regulatory element without traffic light. It refers to the line string without
+  // traffic light attribute.
+  tl_reg_elem_no_tl = TrafficLight::make(
+    99999, tl_re_attr, {ls},
+    {LineString3d(
+      getId(), {Point3d(getId(), 3.0, 3.0, 0.1), Point3d(getId(), 3.0, 4.0, 0.1)}, sl_attr)});
+  test_map_ptr->add(tl_reg_elem_no_tl);
+  addTestMap(test_map_ptr);
+
+  lanelet::validation::RegulatoryElementsDetailsForTrafficLightsValidator checker_;
```

Comment:
Also modify here, please.
```suggestion
  lanelet::validation::RegulatoryElementsDetailsForTrafficLightsValidator checker;
```

Reply:
I'm sorry for missing it.
Fixed in here https://github.com/autowarefoundation/autoware_tools/pull/118/commits/2c9dca1ae149806c8039367f9a63ca259ae30b39